src/main/java/de/tum/in/www1/artemis/web/rest/StudentQuestionResource.java

```diff
@@ -200,6 +205,28 @@ public StudentQuestionResource(StudentQuestionRepository studentQuestionReposito
         return new ResponseEntity<>(studentQuestions, null, HttpStatus.OK);
     }
 
+    /**
+     *
+     * GET /courses/{courseId}/student-questions : get all student questions for course.
+     * @param courseId the course that the student questions belong to
+     * @return the ResponseEntity with status 200 (OK) and with body all student questions for exercise
+     */
+    @GetMapping("courses/{courseId}/student-questions")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<StudentQuestion>> getAllQuestionsForCourse(@PathVariable Long courseId) {
+        final User user = userService.getUserWithGroupsAndAuthorities();
+        Optional<Course> course = courseRepository.findById(courseId);
+        if (course.isEmpty()) {
+            throw new EntityNotFoundException("Course with courseId " + courseId + " does not exist!");
+        }
+        if (!authorizationCheckService.isAtLeastTeachingAssistantInCourse(course.get(), user)) {
+            return forbidden();
+        }
+        List<StudentQuestion> studentQuestions = studentQuestionService.findStudentQuestionsForCourse(courseId);
+
```

Comment:
The other similar calls use `hideSensitiveInformation` here. Was that intentionally left out for the course questions?

Reply:
@FrankeLukas  This is mostly used to hide information students shouldn't be able to see. In this case no such information is sent and the endpoint is only accessible for tutors and instructors.